test/Npgsql.Tests/MultipleHostsTests.cs

```diff
@@ -283,6 +283,81 @@ public void TargetSessionAttributes_default_is_Any()
         {
             var builder = new NpgsqlConnectionStringBuilder();
             Assert.That(builder.TargetSessionAttributes, Is.EqualTo(TargetSessionAttributes.Any.ToString()));
+            Assert.That(builder.TargetSessionAttributesParsed, Is.EqualTo(TargetSessionAttributes.Any));
+        }
+
+        [Test]
+        public void TargetSessionAttributes_invalid_is_Any()
```

Comment:
When I do not understand what exactly was the point of making it a string other that making it tolerable to invalid values.

Reply:
The point was to allow future values (that don't yet exist today), which may also co-exist with primary/standby settings. Another reason was compatibility with libpq, where this is a string.

That doesn't mean that users should be able to write an invalid value - is there any reason to allow this? Imagine a user writing `Target Session Attributes=Stanby` and getting a primary because of the typo...

Reply:
> The point was to allow future values (that don't yet exist today), which may also co-exist with primary/standby settings. Another reason was compatibility with libpq, where this is a string.

That would still throw an exception unless it's handled by us, no? Which means, upgrading to the latest version of npgsql. Which doesn't differ from enum.

Reply:
Yeah, guess it's my mistake when. Will take a look at libpq.